### PR TITLE
fix slo

### PR DESCRIPTION
**Pull request by TheoBrigitte**
I believe we only need to 1 to 1 matching and not 1 to many/many to 1.

As on one side we have ...

image

... and this on the other side

image

So we are matching each `slo_errors_per_request` with its corresponding `slo_threshold`.

### CODE REVIEW

**Review by ubergesundheit — APPROVED** [review]
LGTM (didn't test)

File: helm/prometheus-rules/templates/recording-rules/service-level.rules.yml [hunk 1]
```diff
@@ -238,7 +238,7 @@ spec:
         class: MEDIUM
       record: raw_slo_errors
       # -- 99% availability
-    - expr: "vector((1 - 0.99))"
+    - expr: label_replace(group(rest_client_requests_total{app=~"kube-controller-manager|kube-scheduler"}) by (cluster_id, app), "service", "$1", "app", "(.*)") * 0 + 1 - 0.99
```
Comment (TheoBrigitte) [inline helm/prometheus-rules/templates/recording-rules/service-level.rules.yml:241]: Keeping 99% availability but adding the missing labels

image

File: helm/prometheus-rules/templates/recording-rules/service-level.rules.yml [hunk 2]
```diff
@@ -259,7 +259,7 @@ spec:
         provider: azure
       record: raw_slo_errors
       # -- 99% availability
-    - expr: "vector((1 - 0.99))"
+    - expr: label_replace(group(cloudprovider_azure_api_request_duration_seconds_count{app=~"kube-controller-manager|kube-scheduler"}) by (cluster_id, app), "service", "$1", "app", "(.*)") * 0 + 1 - 0.99
```
Comment (TheoBrigitte) [inline helm/prometheus-rules/templates/recording-rules/service-level.rules.yml:262]: Keeping 99% availability but adding the missing and mandatory `service` label

image

File: helm/prometheus-rules/templates/recording-rules/service-level.rules.yml [hunk 3]
```diff
@@ -281,7 +281,7 @@ spec:
         provider: aws
       record: raw_slo_errors
       # -- 99% availability
-    - expr: "vector((1 - 0.99))"
+    - expr: label_replace(group(cloudprovider_aws_api_request_duration_seconds_count{app=~"kube-controller-manager|kube-scheduler"}) by (cluster_id, app), "service", "$1", "app", "(.*)") * 0 + 1 - 0.99
```
Comment (TheoBrigitte) [inline helm/prometheus-rules/templates/recording-rules/service-level.rules.yml:284]: Keeping 99% availability but adding the missing and mandatory `service` label

image

**Comment by TheoBrigitte** [comment]
lol now we have a conflict because we monitor `kube-controller-manager` twice

```
level=warn ts=2022-02-01T09:26:27.088Z caller=manager.go:603 component="rule manager" group=service-level msg="Evaluating rule failed" rule="alert: ServiceLevelBurnRateTooHigh\nexpr: (slo_errors_per_request:ratio_rate1h{service!~\"efk-.*|.*external-dns.*|kong-.*|.*nginx-.*\"}\n  > on(service) slo_threshold_high and slo_errors_per_request:ratio_rate5m{service!~\"efk-.*|.*external-dns.*|kong-.*|.*nginx-.*\"}\n  > on(service) slo_threshold_high) or (slo_errors_per_request:ratio_rate6h{service!~\"efk-.*|.*external-dns.*|kong-.*|.*nginx-.*\"}\n  > on(service) slo_threshold_low and slo_errors_per_request:ratio_rate30m{service!~\"efk-.*|.*external-dns.*|kong-.*|.*nginx-.*\"}\n  > on(service) slo_threshold_low)\nfor: 5m\nlabels:\n  cancel_if_cluster_has_no_workers: \"true\"\n  cancel_if_cluster_status_creating: \"true\"\n  cancel_if_cluster_status_deleting: \"true\"\n  cancel_if_cluster_status_updating: \"true\"\n  severity: page\nannotations:\n  description: Service level burn rate is too high for {{ $labels.service }} service.\n  opsrecipe: service-level-burn-rate-too-high/\n" err="found duplicate series for the match group {service=\"kube-controller-manager\"} on the right hand-side of the operation: [{__name__=\"slo_threshold_high\", app=\"kube-controller-manager\", area=\"kaas\", cluster_id=\"gauss\", service=\"kube-controller-manager\"}, {__name__=\"slo_threshold_high\", app=\"kube-controller-manager\", area=\"kaas\", cluster_id=\"gauss\", provider=\"aws\", service=\"kube-controller-manager\"}];many-to-many matching not allowed: matching labels must be unique on one side"
```

**Comment by TheoBrigitte** [comment]
To be tested ...

**Comment by TheoBrigitte** [comment]
Tested and working on `gremlin` :heavy_check_mark: 

image

image

**Pull request merged**